tic_tac_toe/src/board.py

```diff
@@ -0,0 +1,203 @@
+# Board Representation
+
+# test_board = [0, 1, -1, 0, 1, -1, 0, 1, -1]
+# 9 values, for 9 spaces on the board
+#
+# -1 = X
+# 0 = No Value
+# 1 = O
+#
+
+
+class Board:
+    def __init__(self, dim):
+        self.values = [None, None, None, None, None, None, None, None, None]
+
+        self.template = "{}|{}|{}\n-------\n{}|{}|{}\n-------\n{}|{}|{}"
+
+        self.dim = dim
+
+        self.winner = None
+
+    def convert(self, x, y):
+        """
+        converts a 2-dimensional coordinate to a 1-dimensional coordinate
+        Parameters
+        ----------
+        x : integer or float
+        the first of the two 2-dim coordinates to be operated on
+
+        y : integer or float
+        the first of the two 2-dim coordinates to be operated on
+
+        Returns
+        -------
+        integer or float : the final 1-dim coordinate
+        """
+        if x < 0 or y < 0:
+            return False
+
+        elif x > 3 or y > 3:
+            return False
+
+        else:
+            return ((y - 1) * int(self.dim)) + (x - 1)
+
+    def render_board(self):
+        """
+        Renders the board in a way that the user can easily read it
+        Returns
+        -------
+        string : The board rendered with an actual tic tac toe board template
+        """
+
+        # Print out the board, in a way humans can understand
+        return self.template.format(*self.values)
+
+    def validate_move(self, x, y, player):
+        """
+        Takes in a move as 2-dim coordinates, and a player, and validates that it is a valid move
+        Parameters
+        ----------
+        x : integer or float
+        The first of the two 2-dim coordinates to be operated on
+
+        y : integer or float
+        The first of the two 2-dim coordinates to be operated on
+
+        player : string
+        "X" or "O", which player
+        Returns
+        ---------
+        : boolean
+        Whether or not it is a valid move
+        """
+        _index = self.convert(x, y)
+
+        if self.values[_index] == None:
+            return True
+
+        elif self.values[_index] == "X":
+            if player == "X":
+                return False
+
+            elif player == "O":
+                return True
+
+        elif self.values[_index] == "O":
+            if player == "O":
+                return False
+
+            elif player == "X":
+                return True
+
+    def check_column(self, board, x_or_o):
+        """
+        checks if a specific player has won with a column win
+        parameters
+        ----------
+        board : class
+        the board to operate on
+
+        x_or_o : string
+        which player we are checking for
+
+        returns
+        ----------
+        winner : boolean
+        whether or not that player won with a column win
+        """
+        board_width = 3
+        winner = None
```

Comment:
Think deeper about the default here.  Should it be `None`?

Reply:
Yes, I believe so

Reply:
So `winner` is true, false or None.  
* How should I interpret `None` in this case?  
* Perhaps in the case when it's None at the end of this function, you throw?  
* Under what conditions would it be None?

Reply:
Actually, because of the way I use `check_winner`, it is alot of the times just `None` because wrong inputs. So im not going to throw at the end if the final product is `None`

Reply:
Ok, fine.  It's weird to mix None's and booleans though.